ext/lmdb-safe/lmdb-safe.cc

```diff
@@ -42,7 +42,9 @@ namespace LMDBLS {
 
     size_t headersize = LS_MIN_HEADER_SIZE;
 
-    headersize += ntohs(lsh->d_numextra) * LS_BLOCK_SIZE;
+    uint16_t numextra = (val[LS_NUMEXTRA_OFFSET] << 8) + val[LS_NUMEXTRA_OFFSET+1];
```

Comment:
As `string_view` holds to chars, the potential signedness can destroy your values:
```
        char a = 0xff;
        char b = 0xff;
        unsigned short x = (a << 8) + b;
        printf("%hx\n", x);
```
Will print `0xffef` if `char` is signed. 
USe
`uint16_t numextra = ((uint16_t)val[LS_NUMEXTRA_OFFSET] << 8) + (uint16_t)val[LS_NUMEXTRA_OFFSET+1];`